var/spack/repos/builtin/packages/mercury/package.py

```diff
@@ -65,6 +65,7 @@ class Mercury(CMakePackage):
     with when("+ofi"):
         depends_on("libfabric@1.5:", when="@:2.0.1")
         depends_on("libfabric@1.7:", when="@2.1.0:")
+        conflicts("libfabric@2:")
```

Comment:
Apologies for the slow response, could you please add that instead, since we support libfabric 2.x in main/master branch.
```suggestion
        conflicts("libfabric@2:", when="@:2.4.0")
```